tests/framework/CodeInterfaceTests/gold/GenericDataSet/samples.csv

```diff
@@ -0,0 +1,5 @@
+autoindex,power,x,y,z,temperature
+0,100.0,1.0,1.0,1.0,3.0
+1,100.0,2.0,1.0,1.0,5.0
+0,200.0,1.0,1.0,1.0,8.0
+1,200.0,2.0,1.0,1.0,10.0
```

Comment:
From this results, it seems to me the autoindex  is used to label the index in each sampled output, i.e., the index for each row in sample/1/All.csv. If this is the case, it seems it is inconsistent for us to treat the samples. For raven, each sample file is one realization, but in this case, each sample file contains multiple realizations. Could you comment on this?

Reply:
This is useful for supporting cases where each sample outputs multiple data points, such as a temperature for each x,y,z location.

Reply:
I'm thinking to have two separate indices, one is the "RAVEN_Sample_ID", which is automatically generated, and the other is autoindex. From this gold file, it seems only autoindex is used. I would like to add  "RAVEN_Sample_ID" (or prefix). In this, we will have two indices that uniquely determine each row of data. Please correct me if I'm wrong.

Reply:
I would suggest we add it.

Reply:
Actually, we do have an automatically generated sample index already.  It is called 'CURRENT_ID' so if the dataset is:
```xml
    <DataSet name="samples">
      <Input>power, autoindex</Input>
      <Output>x, y, z, temperature, CURRENT_ID</Output>
      <Index var="autoindex" autogenerate="True">x, y, z, temperature</Index>
    </DataSet>
```
we get this csv:
```
autoindex,power,x,y,z,temperature,CURRENT_ID
0,100.0,1.0,1.0,1.0,3.0,0
1,100.0,2.0,1.0,1.0,5.0,0
0,200.0,1.0,1.0,1.0,8.0,1
1,200.0,2.0,1.0,1.0,10.0,1
```
This is sort of an undocumented feature (we use CURRENT_ID internally)
Should this be documented? Should I add this to the test?

Reply:
@wangcj05 We have RAVEN_sample_ID in various places, but that currently is not accessible when creating the dataset (without code additions).  prefix also works and might be better than CURRENT_ID.  Let me know how you would like to solve this.

Reply:
I prefere prefix since it is used more often than CURRENT_ID.

Reply:
I added prefix to the input file and the gold file.

Reply:
Thanks a lot @joshua-cogliati-inl. Should we update the manual to explain it? what do you think?

Reply:
I am not sure.  We could probably mention it in the data objects part of the manual.

Reply:
I mentioned prefix in the manual.